### PR TITLE
feat(derive): Signal Receiving

### PR DESCRIPTION
**Pull request by refcell**
### Description

Updates the Derivation Pipeline to coalesce the `FlushableStage` and `ResettableStage` traits into a single `SignalReceiver` trait that accepts the full `Signal`. This also breaks the `signal` method out of the `Pipeline` trait, implementing the `SignalReceiver` on the `DerivationPipeline` as well.

This allows stages to interpret signals themselves, what I originally intended with https://github.com/anton-rs/kona/pull/565.

### CODE REVIEW

File: crates/derive/src/stages/batch_queue.rs [hunk 1]
```diff
@@ -459,45 +459,40 @@ where
...
+            s @ Signal::HoloceneActivation { .. } => {
+                self.prev.signal(s).await?;
+                self.batches.clear();
+                self.next_spans.clear();
+            }
+            Signal::FlushChannel => {
+                self.batches.clear();
+                self.next_spans.clear();
+                self.prev.signal(Signal::FlushChannel).await?;
+            }
```
Comment (clabby) [inline crates/derive/src/stages/batch_queue.rs:490]: Equivalent - can we merge?

**Review by clabby — COMMENTED** [review]
Very nice change. Will approve once CI is green

**Comment by refcell** [comment]
~Aw man, merge conflicts~

Oh yay code updates ✨

**Review by clabby — APPROVED** [review]
very clean mr wick

**Pull request merged**